### ISSUE TITLE
Catch sqlalchemy.exc.IntegrityError

### ISSUE THREAD

**Issue by arnikz** [RDFLib/rdflib-sqlalchemy#35]
Hi,

I would like to catch `IntegrityError` in the presence of potential duplicates (triples). My current env:

rdflib-sqlalchemy (0.3.8)
rdflib (4.2.2)
SQLAlchemy (1.1.9)

```
from __future__ import print_function
from rdflib import plugin, Graph, Literal, URIRef
from rdflib.store import Store
from rdflib_sqlalchemy import registerplugins
from sqlalchemy.exc import IntegrityError

registerplugins()
identifier = URIRef("test")
db_uri = Literal('sqlite:///test.db')
store = plugin.get("SQLAlchemy", Store)(identifier=identifier, configuration=db_uri)
g = Graph(store)
s1=URIRef('http://example.com/Volvo')
s2=URIRef('http://example.com/VW'
p=URIRef('http://example.com/is-a')
o=URIRef('http://example.com/car'))
for s,p,o in ((s1,p,o),(s1,p,o),(s2,p,o)):
   try:
      g.add((s,p,o))
   except IntegrityError:
      pass
print(g.serialize(format='turtle'))
```
```
ERROR:rdflib_sqlalchemy.store:Add failed with statement: INSERT INTO kb_a94a8fe5cc_asserted_statements (id, subject, predicate, object, context, termcomb) VALUES (:id, :subject, :predicate, :object, :context, :termComb), params: {'termComb': 1, 'predicate': rdflib.term.URIRef(u'http://example.com/is-a'), 'object': rdflib.term.URIRef(u'http://example.com/car'), 'context': rdflib.term.BNode('N6c8b02c004c54e18b5712f2aef28a567'), 'subject': rdflib.term.URIRef(u'http://example.com/Volvo')}
Traceback (most recent call last):
  File "/home/arni/.testenv/lib/python2.7/site-packages/rdflib_sqlalchemy/store.py", line 287, in add
    connection.execute(statement, params)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 945, in execute
    return meth(self, multiparams, params)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 263, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1053, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
    context)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
    exc_info
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/home/arni/.testenv/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: kb_a94a8fe5cc_asserted_statements.subject, kb_a94a8fe5cc_asserted_statements.predicate, kb_a94a8fe5cc_asserted_statements.object, kb_a94a8fe5cc_asserted_statements.context [SQL: u'INSERT INTO kb_a94a8fe5cc_asserted_statements (subject, predicate, object, context, termcomb) VALUES (?, ?, ?, ?, ?)'] [parameters: (u'http://example.com/Volvo', u'http://example.com/is-a', u'http://example.com/car', u'N6c8b02c004c54e18b5712f2aef28a567', 1)]
>>> print(g.serialize(format='turtle'))
@prefix ns1: <http://example.com/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

ns1:VW ns1:is-a ns1:car .

ns1:Volvo ns1:is-a ns1:car .
```

**Comment by PonteIneptique** [comment]
Hi There !
I would really much like to do the same, as I have not the hand on the tool that generates the duplicates... Did you end up succeeding @arnikz  ?

**Comment by bovlb** [comment]
Yes.  There is no indication in the documentation for graph.add() that duplicate triples are not allowed, and this is not the behaviour of the in-memory graph.  So far as I can see, all stores ought to tolerate the addition of duplicate triples.  Catching this exception seems like an efficient way to support the correct behaviour.

**Comment by mwatts15** [comment]
This should be resolved with #42 . The change will be included in the next release.
